File: src/main/kotlin/com/huawei/cangjie/psi/CjConstructorDelegationReferenceExpression.java
package com.huawei.cangjie.psi;

import com.huawei.cangjie.lexer.CjTokens;
import com.intellij.lang.ASTNode;
import org.jetbrains.annotations.NotNull;


public class CjConstructorDelegationReferenceExpression extends CjExpressionImpl implements CjReferenceExpression {
    public CjConstructorDelegationReferenceExpression(@NotNull ASTNode node) {
        super(node);
    }
@NotNull
    public boolean isThis() {
        return findChildByType(CjTokens.THIS_KEYWORD) != null;
    }
}

File: src/main/kotlin/com/huawei/cangjie/psi/CjConstructorDelegationReferenceExpression.kt
package com.huawei.cangjie.psi

import com.huawei.cangjie.lexer.CjTokens
import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElement

class CjConstructorDelegationReferenceExpression(node: ASTNode) : CjExpressionImpl(node),
    CjReferenceExpression {
    val isThis: Boolean
        get() = findChildByType<PsiElement?>(CjTokens.THIS_KEYWORD) != null
}
